src/Lecture3.hs

```diff
@@ -246,4 +291,5 @@ Just [8,9,10]
 [8,20,3]
 
 -}
-apply = error "TODO"
+apply :: Applicative f => a -> f (a -> b) -> f b
+apply e f = f <*> pure e
```

Comment:
I see your cheat solution now 👀 

The problem with this implementation is that it requires working with `Applicative`. But not every data type has an `Applicative` instance. So a relaxed version of this function with the `Functor` constraint will be more generally applicable.

Reply:
fixed. also applied your earlier suggestion to eta-reduce to move from `apply a fb = fmap (\f -> f a) fb` to `apply a = fmap (\f -> f a)`.